src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -513,6 +513,61 @@ void SQLStorage::clearMetadata() {
   }
 }
 
+void SQLStorage::storeDelegation(const std::string& data, const Uptane::Role role) {
+  SQLite3Guard db = dbConnection();
+
+  if (!db.beginTransaction()) {
+    LOG_ERROR << "Can't start transaction: " << db.errmsg();
+    return;
+  }
+
+  auto del_statement = db.prepareStatement<std::string>("DELETE FROM delegations WHERE role_name=?;", role.ToString());
+
+  if (del_statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't clear delegation metadata: " << db.errmsg();
+    return;
+  }
+
+  auto ins_statement = db.prepareStatement<SQLBlob, std::string>("INSERT INTO delegations VALUES (?, ?);",
+                                                                 SQLBlob(data), role.ToString());
+
+  if (ins_statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't add delegation metadata: " << db.errmsg();
+    return;
+  }
+
+  db.commitTransaction();
+}
+
+bool SQLStorage::loadDelegation(std::string* data, const Uptane::Role role) {
+  SQLite3Guard db = dbConnection();
+
+  auto statement = db.prepareStatement<std::string>(
+      "SELECT meta FROM delegations WHERE role_name=? ORDER BY role_name DESC LIMIT 1;", role.ToString());
```

Comment:
order by role_name when role_name is fixed?

Also, role_name has a unique constraint, so there should only be one, so no need to `LIMIT 1`.

Reply:
I just normally put `LIMIT 1` everywhere to make it clear that we expect exactly one value (the constraints are not visible in the code). Do you think it's a bad style or can have performance implications?

Reply:
Yes ok that's a fair point

Reply:
Well, I changed it according to @lbonn 's initial comment before I saw @OYTIS 's. Should I restore the `LIMIT 1`? IIRC you can't have that without the `ORDER BY`, though. Alternatively, we could programmatically verify that the statement only returned one row.

Reply:
I've restored `LIMIT 1` without the `ORDER BY` bit.